code/game/machinery/firealarm.dm

```diff
@@ -157,28 +158,46 @@
 	if(machine_stat & NOPOWER)
 		return
 
-	. += mutable_appearance(icon, "fire_overlay")
-	if(is_station_level(z))
-		. += mutable_appearance(icon, "fire_[SSsecurity_level.get_current_level_as_number()]")
-		. += emissive_appearance(icon, "fire_[SSsecurity_level.get_current_level_as_number()]", src, alpha = src.alpha)
-	else
-		. += mutable_appearance(icon, "fire_[SEC_LEVEL_GREEN]")
-		. += emissive_appearance(icon, "fire_[SEC_LEVEL_GREEN]", src, alpha = src.alpha)
-
-	if(!(my_area?.fire || LAZYLEN(my_area?.active_firelocks)))
-		if(my_area?.fire_detect) //If this is false, leave the green light missing. A good hint to anyone paying attention.
-			. += mutable_appearance(icon, "fire_off")
-			. += emissive_appearance(icon, "fire_off", src, alpha = src.alpha)
-	else if(obj_flags & EMAGGED)
-		. += mutable_appearance(icon, "fire_emagged")
-		. += emissive_appearance(icon, "fire_emagged", src, alpha = src.alpha)
-	else
-		. += mutable_appearance(icon, "fire_on")
-		. += emissive_appearance(icon, "fire_on", src, alpha = src.alpha)
+	if(panel_open)
+		return
 
-	if(!panel_open && my_area?.fire_detect && my_area?.fire) //It just looks horrible with the panel open
-		. += mutable_appearance(icon, "fire_detected")
-		. += emissive_appearance(icon, "fire_detected", src, alpha = src.alpha) //Pain
+	if(obj_flags & EMAGGED)
+		. += mutable_appearance(icon, "fire_emag")
+		. += emissive_appearance(icon, "fire_emag_e", src, alpha = src.alpha)
+		set_light(l_color = LIGHT_COLOR_BLUE)
+
+	else if(!(my_area?.fire || LAZYLEN(my_area?.active_firelocks)))
+		if(my_area?.fire_detect) //If this is false, someone disabled it. Leave the light missing, a good hint to anyone paying attention.
+			if(is_station_level(z))
+				var/current_level = SSsecurity_level.get_current_level_as_number()
+				. += mutable_appearance(icon, "fire_[current_level]")
+				. += emissive_appearance(icon, "fire_level_e", src, alpha = src.alpha)
+				switch(current_level)
+					if(SEC_LEVEL_GREEN)
+						set_light(l_color = LIGHT_COLOR_BLUEGREEN)
+					if(SEC_LEVEL_BLUE)
+						set_light(l_color = LIGHT_COLOR_FAINT_BLUE)
+					if(SEC_LEVEL_RED)
+						set_light(l_color = LIGHT_COLOR_FLARE)
+					if(SEC_LEVEL_DELTA)
+						set_light(l_color = COLOR_VIVID_YELLOW)
```

Comment:
ehhhh delta being yellow feels off to me... i get why you chose it like this but it just feels really weird to go "green, blue, red, yellow" because of how commonly we associated 3 out of those 4 colors in traffic stoplights and other general consciousness of "go, slow, stop" corresponding to "green, yellow, red" in specific. taking it out of that order is really odd UX-wise

i don't know what a better color for delta is. the now closed https://github.com/tgstation/tgstation/pull/68702 used a deeper red with a flashing pattern, do you think you can do something like this? Just make it really clear that something's wrong while keeping a deeper red would satisfy me

Reply:
Yeah I can look. Isn't Delta red/yellow alternating on the current fire alarm? That's probably why I did it.